modules/blueBidAdapter.js

```diff
@@ -0,0 +1,106 @@
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { deepSetValue } from '../src/utils.js';
+
+const BIDDER_CODE = 'blue';
+const ENDPOINT_URL = 'https://bidder-us-east-1.getblue.io/engine/?src=prebid';
+const GVLID = 620; // GVLID for your bidder
+const COOKIE_NAME = 'ckid'; // Cookie name for identifying users
+export const storage = getStorageManager({ bidderCode: BIDDER_CODE });
+
+const converter = ortbConverter({
+  context: {
+    netRevenue: true, // Default netRevenue setting
+    ttl: 100, // Default time-to-live for bid responses
+  },
+  imp,
+  request,
+});
+
+function request(buildRequest, imps, bidderRequest, context) {
+  let request = buildRequest(imps, bidderRequest, context);
+  deepSetValue(request, 'site.publisher.id', context.publisherId);
+  // eslint-disable-next-line no-console
+  return request;
+}
+
+function imp(buildImp, bidRequest, context) {
+  let imp = buildImp(bidRequest, context);
+  imp.bidfloor = bidRequest.params.bidFloor;
+  imp.bidfloorcur = bidRequest.params.currency;
+  imp.tagid = bidRequest.params.placementId;
+  return imp;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  supportedMediaTypes: [BANNER], // Supported ad types
+
+  // Validate bid request
+  isBidRequestValid: function (bid) {
+    return (
+      !!bid.params.placementId &&
+      !!bid.params.publisherId &&
+      !!bid.params.bidFloor &&
```

Comment:
You must support the openrtb floor, I would encourage you to simply delete an override floor, but you cannot require one.